test/utils/test_negative_sampling.py

```diff
@@ -31,35 +32,20 @@ def is_negative(edge_index, neg_edge_index, size, bipartite):
 
 def test_edge_index_to_vector_and_vice_versa():
     # Create a fully-connected graph:
-    N = 10
-    row = torch.arange(N).view(-1, 1).repeat(1, N).view(-1)
-    col = torch.arange(N).view(1, -1).repeat(N, 1).view(-1)
+    N1, N2 = 13, 17
+    row = torch.arange(N1).view(-1, 1).repeat(1, N2).view(-1)
+    col = torch.arange(N2).view(1, -1).repeat(N1, 1).view(-1)
     edge_index = torch.stack([row, col], dim=0)
 
-    idx, population = edge_index_to_vector(edge_index, (N, N), bipartite=True)
-    assert population == N * N
-    assert idx.tolist() == list(range(population))
-    edge_index2 = vector_to_edge_index(idx, (N, N), bipartite=True)
-    assert is_undirected(edge_index2)
+    idx = edge_index_to_vector_id(edge_index, (N1, N2))
+    assert idx.tolist() == list(range(N1 * N2))
+    edge_index2 = torch.stack(vector_id_to_edge_index(idx, (N1, N2)), dim=0)
     assert edge_index.tolist() == edge_index2.tolist()
 
-    idx, population = edge_index_to_vector(edge_index, (N, N), bipartite=False)
-    assert population == N * N - N
-    assert idx.tolist() == list(range(population))
-    mask = edge_index[0] != edge_index[1]  # Remove self-loops.
-    edge_index2 = vector_to_edge_index(idx, (N, N), bipartite=False)
-    assert is_undirected(edge_index2)
-    assert edge_index[:, mask].tolist() == edge_index2.tolist()
-
-    idx, population = edge_index_to_vector(edge_index, (N, N), bipartite=False,
-                                           force_undirected=True)
-    assert population == (N * (N + 1)) / 2 - N
-    assert idx.tolist() == list(range(population))
-    mask = edge_index[0] != edge_index[1]  # Remove self-loops.
-    edge_index2 = vector_to_edge_index(idx, (N, N), bipartite=False,
-                                       force_undirected=True)
-    assert is_undirected(edge_index2)
-    assert edge_index[:, mask].tolist() == to_undirected(edge_index2).tolist()
+    vector_id = torch.arange(N1 * N2)
+    edge_index3 = torch.stack(vector_id_to_edge_index(vector_id, (N1, N2)),
+                              dim=0)
+    assert edge_index.tolist() == edge_index3.tolist()
 
 
 def test_negative_sampling():
```

Comment:
Decorate with `@withCUDA` to test on cpu and gpu. 
See [this](https://github.com/pyg-team/pytorch_geometric/blob/241a8c3d018636c116fd1fd7fa2ab9ff3925531e/test/nn/conv/test_nn_conv.py#L13) for an example.